model/map/src/main/java/org/keycloak/models/map/common/locking/ConcurrentMapLocksUtils.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.models.map.common.locking;
+
+import org.keycloak.common.util.Time;
+import org.keycloak.models.locking.LockAcquiringTimeoutException;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.time.Instant;
+import java.util.concurrent.ConcurrentMap;
+ import java.util.function.Supplier;
+
+public class ConcurrentMapLocksUtils {
```

Comment:
Does this belong to `map` module or rather into `map-hot-rod`?

Reply:
See my response to your previous comment.